File: algorithms/easy/special_array_1.py
from typing import List

class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        for i in range(1, len(nums)):
            if (nums[i-1] + nums[i]) % 2 == 0:
                return False
        return True

# Main section
for nums in [
               [1],
               [2,1,4],
               [4,3,1,6],
            ]:
    print(f'nums = {nums}')
    sol = Solution()
    r = sol.isArraySpecial(nums)
    print(f'r = {r}')
    print('======================')

File: algorithms/easy/special_array_1.go
package main

import "fmt"

func isArraySpecial(nums []int) bool {
    for i := 1; i < len(nums); i++ {
        if (nums[i-1] & 1) == (nums[i] & 1) {
            return false
        }
    }
    return true
}

func main() {
    data := [][]int{
                []int{1},
                []int{2,1,4},
                []int{4,3,1,6},
            }
    for _, nums := range data {
        fmt.Printf("nums = %v\n", nums)
        r := isArraySpecial(nums)
        fmt.Printf("r = %v\n", r)
        fmt.Println("=================")
    }
}


